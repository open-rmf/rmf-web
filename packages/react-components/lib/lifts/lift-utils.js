var _a, _b;
import * as RmfModels from 'rmf-models';
export function liftModeToString(liftMode) {
  switch (liftMode) {
    case RmfModels.LiftState.MODE_AGV:
      return 'AGV';
    case RmfModels.LiftState.MODE_EMERGENCY:
      return 'Emergency';
    case RmfModels.LiftState.MODE_FIRE:
      return 'Fire';
    case RmfModels.LiftState.MODE_HUMAN:
      return 'Human';
    case RmfModels.LiftState.MODE_OFFLINE:
      return 'Offline';
    default:
      return 'Unknown (' + liftMode + ')';
  }
}
export function doorStateToString(doorState) {
  switch (doorState) {
    case RmfModels.LiftState.DOOR_OPEN:
      return 'Open';
    case RmfModels.LiftState.DOOR_CLOSED:
      return 'Closed';
    case RmfModels.LiftState.DOOR_MOVING:
      return 'Moving';
    default:
      return 'Unknown (' + doorState + ')';
  }
}
export function motionStateToString(motionState) {
  switch (motionState) {
    case RmfModels.LiftState.MOTION_DOWN:
      return 'Down';
    case RmfModels.LiftState.MOTION_STOPPED:
      return 'Stopped';
    case RmfModels.LiftState.MOTION_UP:
      return 'Up';
    default:
      return 'Unknown (' + motionState + ')';
  }
}
export var requestModes = [
  RmfModels.LiftRequest.REQUEST_AGV_MODE,
  RmfModels.LiftRequest.REQUEST_HUMAN_MODE,
  RmfModels.LiftRequest.REQUEST_END_SESSION,
];
export var requestModeStrings =
  ((_a = {}),
  (_a[RmfModels.LiftRequest.REQUEST_END_SESSION] = 'End Session'),
  (_a[RmfModels.LiftRequest.REQUEST_AGV_MODE] = 'AGV'),
  (_a[RmfModels.LiftRequest.REQUEST_HUMAN_MODE] = 'Human'),
  _a);
export function requestModeToString(requestMode) {
  return requestModeStrings[requestMode] || 'Unknown (' + requestMode + ')';
}
export var requestDoorModes = [RmfModels.LiftRequest.DOOR_OPEN, RmfModels.LiftRequest.DOOR_CLOSED];
export var requestDoorModeStrings =
  ((_b = {}),
  (_b[RmfModels.LiftRequest.DOOR_OPEN] = 'Open'),
  (_b[RmfModels.LiftRequest.DOOR_CLOSED] = 'Closed'),
  _b);
export function requestDoorModeToString(requestDoorMode) {
  return requestDoorModeStrings[requestDoorMode] || 'Unknown';
}
