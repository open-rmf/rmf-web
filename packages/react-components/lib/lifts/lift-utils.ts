import { LiftRequest as RmfLiftRequest, LiftState as RmfLiftState } from 'rmf-models';

export function liftModeToString(liftMode?: number): string {
  if (liftMode === undefined) {
    return `Unknown (${liftMode})`;
  }
  switch (liftMode) {
    case RmfLiftState.MODE_AGV:
      return 'AGV';
    case RmfLiftState.MODE_EMERGENCY:
      return 'Emergency';
    case RmfLiftState.MODE_FIRE:
      return 'Fire';
    case RmfLiftState.MODE_HUMAN:
      return 'Human';
    case RmfLiftState.MODE_OFFLINE:
      return 'Offline';
    default:
      return `Unknown (${liftMode})`;
  }
}

export function doorStateToString(doorState?: number): string {
  if (doorState === undefined) return 'Unknown';

  switch (doorState) {
    case RmfLiftState.DOOR_OPEN:
      return 'OPEN';
    case RmfLiftState.DOOR_CLOSED:
      return 'CLOSED';
    case RmfLiftState.DOOR_MOVING:
      return 'MOVING';
    default:
      return `UNKNOWN (${doorState})`;
  }
}

export function motionStateToString(motionState?: number): string {
  if (motionState === undefined) return 'Unknown';

  switch (motionState) {
    case RmfLiftState.MOTION_DOWN:
      return 'Down';
    case RmfLiftState.MOTION_STOPPED:
      return 'Stopped';
    case RmfLiftState.MOTION_UP:
      return 'Up';
    default:
      return `Unknown (${motionState})`;
  }
}

export const requestModes = [
  RmfLiftRequest.REQUEST_AGV_MODE,
  RmfLiftRequest.REQUEST_HUMAN_MODE,
  RmfLiftRequest.REQUEST_END_SESSION,
];

export const requestModeStrings: Record<number, string> = {
  [RmfLiftRequest.REQUEST_END_SESSION]: 'End Session',
  [RmfLiftRequest.REQUEST_AGV_MODE]: 'AGV',
  [RmfLiftRequest.REQUEST_HUMAN_MODE]: 'Human',
};

export function requestModeToString(requestMode: number): string {
  return requestModeStrings[requestMode] || `Unknown (${requestMode})`;
}

export const requestDoorModes = [RmfLiftRequest.DOOR_OPEN, RmfLiftRequest.DOOR_CLOSED];

export const requestDoorModeStrings: Record<number, string> = {
  [RmfLiftRequest.DOOR_OPEN]: 'Open',
  [RmfLiftRequest.DOOR_CLOSED]: 'Closed',
};

export function requestDoorModeToString(requestDoorMode: number): string {
  return requestDoorModeStrings[requestDoorMode] || 'Unknown';
}
