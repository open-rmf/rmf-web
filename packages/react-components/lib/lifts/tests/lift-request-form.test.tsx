import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { LiftRequestForm } from '..';
import { requestDoorModes, requestModes } from '../lift-utils';
import { makeLift } from './test-utils';

function renderLiftRequestForm() {
  return render(
    <LiftRequestForm
      lift={makeLift()}
      availableRequestTypes={requestModes}
      availableDoorModes={requestDoorModes}
    />,
  );
}

let root: ReturnType<typeof renderLiftRequestForm>;

beforeEach(() => {
  root = render(
    <LiftRequestForm
      lift={makeLift()}
      availableRequestTypes={requestModes}
      availableDoorModes={requestDoorModes}
    />,
  );
});

test('resets form after submitting', () => {
  const initialDestinationHtml = root.getByTestId('destination').outerHTML;
  const initialDoorStateHtml = root.getByTestId('door-state').outerHTML;
  const initialRequestTypeHtml = root.getByTestId('request-type').outerHTML;

  fireEvent.submit(root.getByTestId('request-form'));

  expect(root.getByTestId('destination').outerHTML).toBe(initialDestinationHtml);
  expect(root.getByTestId('door-state').outerHTML).toBe(initialDoorStateHtml);
  expect(root.getByTestId('request-type').outerHTML).toBe(initialRequestTypeHtml);
});

test('destination is required', () => {
  const autocomplete = root.getByTestId('destination');
  const input = autocomplete.querySelector('input');
  if (!input) {
    fail('destination value element not found');
  }
  fireEvent.change(input, {
    target: { value: '' },
  });
  fireEvent.submit(root.getByTestId('request-form'));
  expect(root.container.querySelectorAll('[data-error]')).toHaveLength(1);
});
