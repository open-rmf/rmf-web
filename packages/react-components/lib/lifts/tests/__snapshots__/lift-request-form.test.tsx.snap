// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <form
    class="makeStyles-form-1"
    data-testid="request-form"
  >
    <div
      class="makeStyles-divForm-2"
    >
      <div
        aria-expanded="false"
        class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
        data-testid="destination"
        role="combobox"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink="true"
            for="mui-94122"
            id="mui-94122-label"
          >
            Pick a Destination
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <input
              aria-autocomplete="list"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              id="mui-94122"
              spellcheck="false"
              type="text"
              value="test"
            />
            <div
              class="MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                tabindex="-1"
                title="Clear"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-9 PrivateNotchedOutline-legendNotched-10"
              >
                <span>
                  Pick a Destination
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
      </div>
    </div>
    <div
      class="makeStyles-divForm-2"
    >
      <div
        aria-expanded="false"
        class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
        data-testid="door-state"
        role="combobox"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink="true"
            for="mui-84471"
            id="mui-84471-label"
          >
            Pick a Door State
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <input
              aria-autocomplete="list"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              id="mui-84471"
              spellcheck="false"
              type="text"
              value="Open"
            />
            <div
              class="MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                tabindex="-1"
                title="Clear"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-9 PrivateNotchedOutline-legendNotched-10"
              >
                <span>
                  Pick a Door State
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
      </div>
    </div>
    <div
      class="makeStyles-divForm-2"
    >
      <div
        aria-expanded="false"
        class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
        data-testid="request-type"
        role="combobox"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink="true"
            for="mui-85070"
            id="mui-85070-label"
          >
            Pick Request Type
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <input
              aria-autocomplete="list"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              id="mui-85070"
              spellcheck="false"
              type="text"
              value="End Session"
            />
            <div
              class="MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                tabindex="-1"
                title="Clear"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-9 PrivateNotchedOutline-legendNotched-10"
              >
                <span>
                  Pick Request Type
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
      </div>
    </div>
    <div
      class="makeStyles-buttonContainer-6"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-5 MuiButton-containedPrimary"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Request
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;
